### PR TITLE
fix: incorrect db types on retrieval/insertion

### PR DESCRIPTION
**Pull request by igamigo**
(draft/WIP)

Potential way to fix https://github.com/0xPolygonMiden/miden-node/issues/147. This PR has many ad-hoc changes aimed at unblocking the testing of the client. As such, some changes might require further thought.
As a notable drawback, this removes 2 constraints from the DB which wouldn't directly work considering the underlying type is `i64`.

TODO:
- Use newer genesis accounts generation and remove custom changes

### CODE REVIEW

File: store/src/db/mod.rs [hunk 1]
```diff
@@ -54,22 +54,10 @@ impl Db {
...
-                            // synchronization of the servers in-memory and on-disk representations
-                            // (see [State::apply_block])
-                            conn.execute("PRAGMA journal_mode = WAL;", ())?;
-
-                            // Enable foreign key checks.
-                            conn.execute("PRAGMA foreign_keys = ON;", ())
-                        })
+                        .interact(|conn| array::load_module(conn))
                         .await
                         .map_err(|_| HookError::StaticMessage("Loading carray module failed"))?;
```
Comment (bobbinth) [inline store/src/db/mod.rs:62]: Were these changes causing issues? If not, we should revert these changes.

File: store/src/server/api.rs [hunk 2]
```diff
@@ -127,7 +127,11 @@ impl api_server::Api for StoreApi {
 
         let notes = request.notes;
 
-        let _ = self.state.apply_block(block, nullifiers, accounts, notes).await;
+        let apply_block_status = self.state.apply_block(block, nullifiers, accounts, notes).await;
+
+        if let Err(err) = apply_block_status {
+            println!("Error: {:?}", err);
+        }
```
Comment (bobbinth) [inline store/src/server/api.rs:134]: Was this a temp change? If so, we should revert it and return the error. If we want to add logging, we should probably use the tracer.

File: store/src/genesis.rs
Comment (bobbinth) [inline store/src/genesis.rs:1]: Changes in this file will probably be superseded by #138.

**Comment by igamigo** [comment]
Not needed after #156
